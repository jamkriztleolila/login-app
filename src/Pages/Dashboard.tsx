import React, { useCallback, useState, useEffect } from 'react';
import {
  Container, Grid, Card, CardContent, CardActions, Typography,
  FormControl, FormLabel, Input, Button, Table, Divider, Alert, Stack
} from '@mui/joy';
import { User } from '../Service/DashboardService';
import { dashboardService, authenticationService, USER_ID } from '../instances';

const Dashboard = () => {
  const initialValues = {
    branchId: NaN,
    userName: '',
    firstName: '',
    middleName: '',
    lastName: '',
    position: '',
    password: '',
  };
  const [usersData, setUsersData] = useState<User[]>([]);
  const [username, setUsername] = useState<string | null>();
  const [userForm, setUserForm] = useState(initialValues);

  const [error, setError] = useState({
    branchId: false,
    userName: false,
    password: false,
  });

  const [alertState, setAlertState] = useState({
    show: false,
    message: '',
    title: '',
    errorColor: false,
  });

  const getUsers = useCallback(async () => {
    try {
      const data = await dashboardService.getAllUsers();
      setUsersData(data);
    } catch (err) {
      console.log(err);
    }
  }, []);

  const handleChange = (e: any) => {
    const { name, value } = e.target;
    setUserForm({
      ...userForm,
      [name]: value,
    });
  }

  const resetFields = async (e: any) => {
    setUserForm(initialValues);
    setAlertState({
      ...alertState,
      show: false
    });
  };

  const validateForm = () => {
    const hasMatchBranchId = usersData.find((u) => { return u.branchId === userForm.branchId });
    setError({
      branchId: userForm.branchId.toString() === '' || userForm.branchId.toString() === 'NaN',
      userName: userForm.userName === '',
      password: userForm.password === ''
    });

    if (hasMatchBranchId) {
      setError({
        ...error,
        branchId: true,
      });
      setAlertState({
        show: true,
        message: 'Branch ID already exists.',
        title: 'Error',
        errorColor: true,
      });
      return false;
    }

    const invalidFields = userForm.branchId.toString() === '' || userForm.branchId.toString() === 'NaN' || userForm.userName === '' || userForm.password === '';
    if (invalidFields) {
      setAlertState({
        show: true,
        message: 'Please fill in required fields.',
        title: 'Error',
        errorColor: true,
      });
    }
    return !invalidFields;
  }

  const handleSubmit = async (event: any) => {
    event.preventDefault();
    if(validateForm() === true) {
      setUsersData(prev => [...prev, userForm]);
      setAlertState({
        show: true,
        message: 'Successfully added user!',
        title: 'Success',
        errorColor: false
      });
    }
  }

  const removeRecord = async (id: number) => {
    setUsersData(prev => {
      const idx = prev.findIndex((user) => user.branchId === id);
      prev.splice(idx, 1)
      return [...prev];
    });
  };

  useEffect(() => {
    if (sessionStorage.getItem(USER_ID) ) {
      setUsername(sessionStorage.getItem(USER_ID));
      getUsers();
    } else {
      authenticationService.logout();
    }
  }, [username, getUsers]);

  return (
    <Container>
      <Grid container spacing={2} sx={{ marginTop: '1em' }}>
        <Grid xs={8}>
          <Typography level="h2">{username}</Typography>
        </Grid>
        <Grid xs={4} justifyContent={'flex-end'}>
          <Button size="lg" 
            onClick={() => authenticationService.logout()} 
            sx={{ marginLeft: 'auto', display: 'block' }} aria-label="logoutBtn">Logout</Button>
        </Grid>
      </Grid>
      <Divider sx={{ my: 1 }} />
      <Grid container spacing={2}>
      <Grid md={4} xs={12}>
        <Card>
          <CardContent>
            <Typography level="title-lg">Add user</Typography>
            <form id="dashboard-form" onSubmit={handleSubmit} autoComplete="false">
              <Stack spacing={1}>
                <FormControl sx={{ gridColumn: '1/-1' }}>
                  <FormLabel required>Branch ID</FormLabel>
                  <Input
                    placeholder="Branch ID"
                    name="branchId"
                    size="sm"
                    onChange={handleChange}
                    value={userForm.branchId}
                    type="number"
                    error={error.branchId}
                    slotProps={{ input: { "aria-label": "branchId" } }} />
                </FormControl>
                <FormControl sx={{ gridColumn: '1/-1' }}>
                  <FormLabel required>Username</FormLabel>
                  <Input
                    placeholder="Username"
                    name="userName"
                    size="sm"
                    onChange={handleChange}
                    value={userForm.userName}
                    error={error.userName}
                    slotProps={{ input: { "aria-label": "username" } }}  />
                </FormControl>
                <FormControl sx={{ gridColumn: '1/-1' }}>
                  <FormLabel>First Name</FormLabel>
                  <Input
                    placeholder="First Name"
                    name="firstName"
                    size="sm"
                    onChange={handleChange}
                    value={userForm.firstName}
                    slotProps={{ input: { "aria-label": "firstName" } }} />
                </FormControl>
                <FormControl sx={{ gridColumn: '1/-1' }}>
                  <FormLabel>Middle Name</FormLabel>
                  <Input
                    placeholder="Middle Name"
                    name="middleName"
                    size="sm"
                    onChange={handleChange}
                    value={userForm.middleName}
                    slotProps={{ input: { "aria-label": "middleName" } }} />
                </FormControl>
                <FormControl sx={{ gridColumn: '1/-1' }}>
                  <FormLabel>Last Name</FormLabel>
                  <Input
                    placeholder="Last Name"
                    name="lastName"
                    size="sm"
                    onChange={handleChange}
                    value={userForm.lastName}
                    slotProps={{ input: { "aria-label": "lastName" } }} />
                </FormControl>
                <FormControl sx={{ gridColumn: '1/-1' }}>
                  <FormLabel>Position</FormLabel>
                  <Input
                    placeholder="Position"
                    name="position"
                    size="sm"
                    onChange={handleChange}
                    value={userForm.position}
                    slotProps={{ input: { "aria-label": "position" } }} />
                </FormControl>
                <FormControl sx={{ gridColumn: '1/-1' }}>
                  <FormLabel required>Password</FormLabel>
                  <Input
                    placeholder="Password"
                    type="password"
                    size="sm"
                    name="password"
                    onChange={handleChange}
                    value={userForm.password}
                    error={error.password}
                    slotProps={{ input: { "aria-label": "password" } }} />
                </FormControl>

                {alertState.show ? <Alert
                    sx={{ alignItems: 'flex-start' }}
                    variant="soft"
                    color={alertState.errorColor ? 'danger' : 'success'}
                    title={alertState.title}>
                    <Typography level="body-sm" color={alertState.errorColor ? 'danger' : 'success'}>
                      {alertState.message}
                    </Typography>
                  </Alert> : ''
                }
                <CardActions sx={{ gridColumn: '1/-1', marginLeft: 'auto !important' }} >
                  <Button variant="outlined" color="primary" size="md" onClick={resetFields} aria-label="resetForm">
                    Reset
                  </Button>
                  <Button variant="solid" color="primary" size="md" type="submit" aria-label="addUser">
                    Add
                  </Button>
              </CardActions>
            </Stack>
          </form>
          </CardContent>
        </Card>
      </Grid>
      <Grid md={8} xs={12}>
      <Table>
        <thead>
          <tr>
            <th style={{ width: '5%' }}>#</th>
            <th style={{ width: '10%' }}>Branch ID</th>
            <th>Name</th>
            <th>Position</th>
            <th>Action</th>
          </tr>
        </thead>
        <tbody>
          {usersData.map((item: User, i: number) => (
            <tr key={`item-${item.branchId}`}>
              <td>{i + 1}</td>
              <td>{item.branchId}</td>
              <td>{`${item.firstName} ${item.middleName} ${item.lastName}`}</td>
              <td>{item.position}</td>
              <td><Button
                color="danger"
                onClick={() => removeRecord(item.branchId)}
                aria-label={`remove-${item.branchId}`}
                >Remove</Button></td>
            </tr>
          ))}
        </tbody>
    </Table>
      </Grid>
      </Grid>
    </Container>
  );
};

export default Dashboard;