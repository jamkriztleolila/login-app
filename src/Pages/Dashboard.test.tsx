import React from 'react';
import { render, screen, renderHook, fireEvent } from '@testing-library/react';
import Dashboard from './Dashboard';
import userEvent from '@testing-library/user-event';
import '../setupTests';


const mockUser = 'testUser01';
const MOCK_USER_ID = 'userId';
const setup = () => {
  const utils = render(<Dashboard />)
  const branchId = screen.getByLabelText('branchId', {selector: 'input'})
  const username = screen.getByLabelText('username', {selector: 'input'})
  const firstName = screen.getByLabelText('firstName', {selector: 'input'})
  const middleName = screen.getByLabelText('middleName', {selector: 'input'})
  const lastName = screen.getByLabelText('lastName', {selector: 'input'})
  const position = screen.getByLabelText('position', {selector: 'input'})
  const password = screen.getByLabelText('password', {selector: 'input'})
  return {
    branchId,
    username,
    firstName,
    middleName,
    lastName,
    position,
    password,
    ...utils,
  }
}

describe('<Dashboard />', () => {
  test('match snapshot', () => {
    const { container } = render(<Dashboard />);
    expect(container).toMatchSnapshot()
  });

  test('read username on dashboard', () => {
    window.sessionStorage.setItem(MOCK_USER_ID, mockUser);
    render(<Dashboard />);
    const usernameOnScreen = screen.getByText(mockUser);
    expect(usernameOnScreen).toBeInTheDocument();
  });

  test('no userId found in sessionStorage', () => {
    render(<Dashboard />);
    if (window.sessionStorage.getItem(MOCK_USER_ID)) {
      window.sessionStorage.removeItem(MOCK_USER_ID);
    }
    expect(window.sessionStorage.getItem(MOCK_USER_ID)).toBeNull();
  });

  describe('handleChange()', () => {
    beforeEach(() => {
      window.sessionStorage.setItem(MOCK_USER_ID, mockUser);
    });
    test('should update state', () => {
      const { branchId } = setup();
      fireEvent.change(branchId, {target: {value: '10001'}})
      expect(branchId).toHaveValue(10001)
    });
  });

  describe('handleSubmit()', () => {
    beforeEach(() => {
      window.sessionStorage.setItem(MOCK_USER_ID, mockUser);
    });
    test('empty fields', async () => {
      render(<Dashboard />);
      await userEvent.click(screen.getByLabelText('addUser', { selector: 'button' }));
      expect(screen.getByText('Please fill in required fields.')).toBeInTheDocument();
    });

    test('branch Id already exists', async () => {
      const { branchId, username, firstName, middleName, lastName, position, password } = setup();
      fireEvent.change(branchId, {target: {value: '10001'}})
      fireEvent.change(username, {target: {value: 'testuser01'}})
      fireEvent.change(firstName, {target: {value: 'John'}})
      fireEvent.change(middleName, {target: {value: 'C'}})
      fireEvent.change(lastName, {target: {value: 'Doe'}})
      fireEvent.change(position, {target: {value: 'QA Tester'}})
      fireEvent.change(password, {target: {value: 'pa55w0rd001'}})
      await userEvent.click(screen.getByLabelText('addUser', { selector: 'button' }));
      await userEvent.click(screen.getByLabelText('addUser', { selector: 'button' }));
      expect(screen.getByText('Branch ID already exists.')).toBeInTheDocument();
    });

    test('success adding user', async () => {
      const { branchId, username, firstName, middleName, lastName, position, password } = setup();
      fireEvent.change(branchId, {target: {value: '20001'}})
      fireEvent.change(username, {target: {value: 'testuser201'}})
      fireEvent.change(firstName, {target: {value: 'Jessa'}})
      fireEvent.change(middleName, {target: {value: 'C'}})
      fireEvent.change(lastName, {target: {value: 'Meek'}})
      fireEvent.change(position, {target: {value: 'Software Tester'}})
      fireEvent.change(password, {target: {value: 'pa55w0rd001'}})
      await userEvent.click(screen.getByLabelText('addUser', { selector: 'button' }));
      expect(screen.getByText('Successfully added user!')).toBeInTheDocument();
    });
  });

  describe('resetFields()', () => {
    beforeEach(() => {
      window.sessionStorage.setItem(MOCK_USER_ID, mockUser);
    });
    test('clear all fields', async () => {
      const { branchId, username, firstName, middleName, lastName, position, password } = setup();
      fireEvent.change(branchId, {target: {value: '20001'}})
      fireEvent.change(username, {target: {value: 'testuser201'}})
      fireEvent.change(firstName, {target: {value: 'Jessa'}})
      fireEvent.change(middleName, {target: {value: 'C'}})
      fireEvent.change(lastName, {target: {value: 'Meek'}})
      fireEvent.change(position, {target: {value: 'Software Tester'}})
      fireEvent.change(password, {target: {value: 'pa55w0rd001'}})
      await userEvent.click(screen.getByLabelText('resetForm', { selector: 'button' }))
      expect(screen.getByLabelText('branchId', {selector: 'input'})).toHaveValue(null)
      expect(screen.getByLabelText('username', {selector: 'input'})).toHaveValue('')
      expect(screen.getByLabelText('firstName', {selector: 'input'})).toHaveValue('')
      expect(screen.getByLabelText('middleName', {selector: 'input'})).toHaveValue('')
      expect(screen.getByLabelText('lastName', {selector: 'input'})).toHaveValue('')
      expect(screen.getByLabelText('position', {selector: 'input'})).toHaveValue('')
      expect(screen.getByLabelText('password', {selector: 'input'})).toHaveValue('')
    });
  });

  describe('removeRecord()', () => {
    beforeEach(() => {
      window.sessionStorage.setItem(MOCK_USER_ID, mockUser);
    });
    test('remove record', async () => {
      const { branchId, username, firstName, middleName, lastName, position, password } = setup();
      fireEvent.change(branchId, {target: {value: '20001'}})
      fireEvent.change(username, {target: {value: 'testuser201'}})
      fireEvent.change(firstName, {target: {value: 'Jessa'}})
      fireEvent.change(middleName, {target: {value: 'C'}})
      fireEvent.change(lastName, {target: {value: 'Meek'}})
      fireEvent.change(position, {target: {value: 'Software Tester'}})
      fireEvent.change(password, {target: {value: 'pa55w0rd001'}})
      await userEvent.click(screen.getByLabelText('addUser', { selector: 'button' }));
      expect(screen.getByText('20001')).toBeInTheDocument();
      await userEvent.click(screen.getByLabelText('remove-20001', { selector: 'button' })); 
      expect(screen.queryByText('20001')).not.toBeInTheDocument();
    });
  });

  describe('removeRecord()', () => { 
    beforeEach(() => {
      window.sessionStorage.setItem(MOCK_USER_ID, mockUser);
    });
    test('logout user', async () => {
      render(<Dashboard />);
      await userEvent.click(screen.getByLabelText('logoutBtn', { selector: 'button' }));
    });
  });
  
});