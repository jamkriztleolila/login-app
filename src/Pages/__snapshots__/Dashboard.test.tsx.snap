// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> match snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-pv8j0-JoyContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-1ifwdrn-JoyGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-8 css-1jigpui-JoyGrid-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-x5jxhy-JoyTypography-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-4 css-2pud3i-JoyGrid-root"
      >
        <button
          aria-label="logoutBtn"
          class="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeLg css-1hh6q5m-JoyButton-root"
          type="button"
        >
          Logout
        </button>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-horizontal css-16a0t0a-JoyDivider-root"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-1ubpz9o-JoyGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-md-4 MuiGrid-grid-xs-12 css-dq3jzk-JoyGrid-root"
      >
        <div
          class="MuiCard-root MuiCard-vertical MuiCard-variantOutlined MuiCard-colorNeutral MuiCard-sizeMd css-1p3o6sg-JoyCard-root"
        >
          <div
            class="MuiCardContent-root css-1wk5k0m-JoyCardContent-root"
            data-first-child=""
            data-last-child=""
          >
            <p
              class="MuiTypography-root MuiTypography-title-lg css-ar9ysw-JoyTypography-root"
            >
              Add user
            </p>
            <form
              autocomplete="false"
              id="dashboard-form"
            >
              <div
                class="MuiStack-root css-1s5gupf-JoyStack-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-vertical MuiFormControl-sizeMd css-zu7ee1-JoyFormControl-root"
                >
                  <label
                    class="MuiFormLabel-root css-1ckvkg0-JoyFormLabel-root"
                    for=":r0:"
                    id=":r0:-label"
                    required=""
                  >
                    Branch ID
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk css-3w38ak-JoyFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInput-root MuiInput-variantOutlined MuiInput-colorNeutral MuiInput-sizeSm MuiInput-formControl css-1534vyt-JoyInput-root"
                  >
                    <input
                      aria-label="branchId"
                      class="MuiInput-input css-1gw9vc6-JoyInput-input"
                      id=":r0:"
                      name="branchId"
                      placeholder="Branch ID"
                      type="number"
                      value="NaN"
                    />
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-vertical MuiFormControl-sizeMd css-zu7ee1-JoyFormControl-root"
                >
                  <label
                    class="MuiFormLabel-root css-1ckvkg0-JoyFormLabel-root"
                    for=":r1:"
                    id=":r1:-label"
                    required=""
                  >
                    Username
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk css-3w38ak-JoyFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInput-root MuiInput-variantOutlined MuiInput-colorNeutral MuiInput-sizeSm MuiInput-formControl css-1534vyt-JoyInput-root"
                  >
                    <input
                      aria-label="username"
                      class="MuiInput-input css-1gw9vc6-JoyInput-input"
                      id=":r1:"
                      name="userName"
                      placeholder="Username"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-vertical MuiFormControl-sizeMd css-zu7ee1-JoyFormControl-root"
                >
                  <label
                    class="MuiFormLabel-root css-1ckvkg0-JoyFormLabel-root"
                    for=":r2:"
                    id=":r2:-label"
                  >
                    First Name
                  </label>
                  <div
                    class="MuiInput-root MuiInput-variantOutlined MuiInput-colorNeutral MuiInput-sizeSm MuiInput-formControl css-1534vyt-JoyInput-root"
                  >
                    <input
                      aria-label="firstName"
                      class="MuiInput-input css-1gw9vc6-JoyInput-input"
                      id=":r2:"
                      name="firstName"
                      placeholder="First Name"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-vertical MuiFormControl-sizeMd css-zu7ee1-JoyFormControl-root"
                >
                  <label
                    class="MuiFormLabel-root css-1ckvkg0-JoyFormLabel-root"
                    for=":r3:"
                    id=":r3:-label"
                  >
                    Middle Name
                  </label>
                  <div
                    class="MuiInput-root MuiInput-variantOutlined MuiInput-colorNeutral MuiInput-sizeSm MuiInput-formControl css-1534vyt-JoyInput-root"
                  >
                    <input
                      aria-label="middleName"
                      class="MuiInput-input css-1gw9vc6-JoyInput-input"
                      id=":r3:"
                      name="middleName"
                      placeholder="Middle Name"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-vertical MuiFormControl-sizeMd css-zu7ee1-JoyFormControl-root"
                >
                  <label
                    class="MuiFormLabel-root css-1ckvkg0-JoyFormLabel-root"
                    for=":r4:"
                    id=":r4:-label"
                  >
                    Last Name
                  </label>
                  <div
                    class="MuiInput-root MuiInput-variantOutlined MuiInput-colorNeutral MuiInput-sizeSm MuiInput-formControl css-1534vyt-JoyInput-root"
                  >
                    <input
                      aria-label="lastName"
                      class="MuiInput-input css-1gw9vc6-JoyInput-input"
                      id=":r4:"
                      name="lastName"
                      placeholder="Last Name"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-vertical MuiFormControl-sizeMd css-zu7ee1-JoyFormControl-root"
                >
                  <label
                    class="MuiFormLabel-root css-1ckvkg0-JoyFormLabel-root"
                    for=":r5:"
                    id=":r5:-label"
                  >
                    Position
                  </label>
                  <div
                    class="MuiInput-root MuiInput-variantOutlined MuiInput-colorNeutral MuiInput-sizeSm MuiInput-formControl css-1534vyt-JoyInput-root"
                  >
                    <input
                      aria-label="position"
                      class="MuiInput-input css-1gw9vc6-JoyInput-input"
                      id=":r5:"
                      name="position"
                      placeholder="Position"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-vertical MuiFormControl-sizeMd css-zu7ee1-JoyFormControl-root"
                >
                  <label
                    class="MuiFormLabel-root css-1ckvkg0-JoyFormLabel-root"
                    for=":r6:"
                    id=":r6:-label"
                    required=""
                  >
                    Password
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk css-3w38ak-JoyFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInput-root MuiInput-variantOutlined MuiInput-colorNeutral MuiInput-sizeSm MuiInput-formControl css-1534vyt-JoyInput-root"
                  >
                    <input
                      aria-label="password"
                      class="MuiInput-input css-1gw9vc6-JoyInput-input"
                      id=":r6:"
                      name="password"
                      placeholder="Password"
                      type="password"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="MuiCardActions-root css-3nynj0-JoyCardActions-root"
                >
                  <button
                    aria-label="resetForm"
                    class="MuiButton-root MuiButton-variantOutlined MuiButton-colorPrimary MuiButton-sizeMd css-wbn1ta-JoyButton-root"
                    type="button"
                  >
                    Reset
                  </button>
                  <button
                    aria-label="addUser"
                    class="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeMd css-cua13l-JoyButton-root"
                    type="submit"
                  >
                    Add
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-md-8 MuiGrid-grid-xs-12 css-1dc0y22-JoyGrid-root"
      >
        <table
          class="MuiTable-root MuiTable-borderAxisXBetween MuiTable-variantPlain MuiTable-colorNeutral MuiTable-sizeMd css-1mbiu7x-JoyTable-root"
        >
          <thead>
            <tr>
              <th
                style="width: 5%;"
              >
                #
              </th>
              <th
                style="width: 10%;"
              >
                Branch ID
              </th>
              <th>
                Name
              </th>
              <th>
                Position
              </th>
              <th>
                Action
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;
